src/heretic/utils.py

```diff
@@ -121,3 +124,83 @@ def get_readme_intro(
 -----
 
 """
+
+
+def is_notebook() -> bool:
+    # Check IPython shell type
+    try:
+        from IPython import get_ipython
+
+        shell = get_ipython()
+        if shell is None:
+            return False
+
+        shell_name = shell.__class__.__name__
+        # Standard Jupyter/Colab shells
+        if shell_name in ["ZMQInteractiveShell", "Shell"]:
+            return True
+
+        # Some other environments might have different shell names but still be notebooks
```

Comment:
I just don't get it. What is the shell used by Colab? It's not random, is it? Above it says that those are the standard Colab shells, and here it says it "might" have a different shell name. What is going on here? Do different Colab notebooks use different shells? Where does this information even come from?

Reply:
Okay, this mystery code really bothered me, so I dug around in some libraries and have found a MUCH better solution:

```python
from rich.console import _is_jupyter as is_notebook
```

That's it! :smiley: We can just get that functionality from a library that Heretic already uses, and the library authors will maintain it for us.

FWIW, they do a somewhat similar (though not identical) thing to what you wrote, and I don't fully understand their version either, but I don't care because I don't have to maintain it.